sdk/keyvault/keyvault-common/test/internal/challengeAuthenticationCallbacks.spec.ts

```diff
@@ -2,279 +2,216 @@
 // Licensed under the MIT License.
 
 import {
-  AuthorizeRequestOptions,
-  ChallengeCallbacks,
+  Pipeline,
   PipelineRequest,
+  SendRequest,
+  createEmptyPipeline,
   createHttpHeaders,
   createPipelineRequest,
 } from "@azure/core-rest-pipeline";
-import { createKeyVaultChallengeCallbacks } from "../../src/index.js";
 import { parseWWWAuthenticateHeader } from "../../src/parseWWWAuthenticate.js";
-import { describe, it, beforeEach, expect } from "vitest";
+import { describe, it, beforeEach, expect, vi } from "vitest";
+import { TokenCredential } from "@azure/core-auth";
+import { keyVaultAuthenticationPolicy } from "../../src/keyVaultAuthenticationPolicy.js";
 
-describe("Challenge based authentication tests", function () {
-  let request: PipelineRequest;
-  let challengeCallbacks: ChallengeCallbacks;
+const caeChallenge = `Bearer realm="", authorization_uri="https://login.microsoftonline.com/common/oauth2/authorize", error="insufficient_claims", claims="eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ=="`;
+const caeClaims = `{"access_token":{"nbf":{"essential":true,"value":"1726077595"},"xms_caeerror":{"value":"10012"}}}`;
 
-  beforeEach(() => {
-    request = createPipelineRequest({ url: "https://myvault.vault.azure.net" });
-    challengeCallbacks = createKeyVaultChallengeCallbacks();
-  });
+const challengeResponse: SendRequest = async (req) => {
+  expect(req.body).toBeNull();
+  return {
+    headers: createHttpHeaders({
+      "WWW-Authenticate": `Bearer resource="https://vault.azure.net", tenantId="testTenantId"`,
```

Comment:
Is this supposed to be a realistic Key Vault challenge? Those look like `Bearer authorization="https://login.microsoftonline.com/tenant", resource="..."`

The tenant ID is the first path segment of the `authorization` URL; there's no `tenantId` parameter